### PR TITLE
Movement: Add both grid based and pixel based movement

### LINKED ISSUE #19: Movement system overhaull
Currently we have a dash mechanic and a pixel based movement how ever if we wish to be combat ready.
we need to have tile based movment system

### PR DESCRIPTION
**Pull request by realstealthninja**

### CODE REVIEW

**Review by nimoooos — CHANGES_REQUESTED** [review]
That's not what we decided on for movement, right? It should be turn based tactics?

**Comment by realstealthninja** [comment]
yep doing that rn. made this pr yesterday

File: project.godot [hunk 1]
```diff
@@ -11,6 +11,12 @@ config_version=5
 [application]
 
 config/name="Prometheus"
+config/features=PackedStringArray("4.2")
```
Comment (realstealthninja) [inline project.godot]:
```suggestion
```

**Comment by realstealthninja** [comment]
DO NOT MERGE YET

**Comment by realstealthninja** [comment]
READY FOR REVIEW

**Review by nimoooos — APPROVED** [review]
I'm a bit confused about the code but I saw the demo so you're good

**Pull request merged**